fieldservice/models/fsm_order.py

```diff
@@ -0,0 +1,426 @@
+    street = fields.Char(related="location_id.street")
+    street2 = fields.Char(related="location_id.street2")
+    zip = fields.Char(related="location_id.zip")
+    city = fields.Char(related="location_id.city")
+    state_name = fields.Char(related="location_id.state_id.name", string="State")
+    country_name = fields.Char(related="location_id.country_id.name", string="Country")
+    phone = fields.Char(related="location_id.phone", string="Location Phone")
+    mobile = fields.Char(related="location_id.mobile")
+
+    stage_name = fields.Char(related="stage_id.name", string="Stage Name")
+    # Field for Stage Color
+    custom_color = fields.Char(related="stage_id.custom_color", string="Stage Color")
+
+    # Template
+    template_id = fields.Many2one("fsm.template", string="Template")
+    category_ids = fields.Many2many("fsm.category", string="Categories")
+
+    # Equipment used for Maintenance and Repair Orders
+    equipment_id = fields.Many2one("fsm.equipment", string="Equipment")
+
+    # Equipment used for all other Service Orders
+    equipment_ids = fields.Many2many("fsm.equipment", string="Equipments")
+    type = fields.Many2one("fsm.order.type")
+
+    internal_type = fields.Selection(related="type.internal_type")
+
+    @api.model
+    def _read_group_stage_ids(self, stages, domain, order):
+        search_domain = [("stage_type", "=", "order")]
+        if self.env.context.get("default_team_id"):
+            search_domain = [
+                "&",
+                ("team_ids", "in", self.env.context["default_team_id"]),
+            ] + search_domain
+        return stages.search(search_domain, order=order)
+
+    @api.model
+    def create(self, vals):
+        if vals.get("name", _("New")) == _("New"):
+            vals["name"] = self.env["ir.sequence"].next_by_code("fsm.order") or _("New")
+        self._calc_scheduled_dates(vals)
+        if not vals.get("request_late"):
+            if vals.get("priority") == "0":
+                if vals.get("request_early"):
+                    vals["request_late"] = fields.Datetime.from_string(
+                        vals.get("request_early")
+                    ) + timedelta(days=3)
+                else:
+                    vals["request_late"] = datetime.now() + timedelta(days=3)
+            elif vals.get("request_early") and vals.get("priority") == "1":
+                vals["request_late"] = fields.Datetime.from_string(
+                    vals.get("request_early")
+                ) + timedelta(days=2)
+            elif vals.get("request_early") and vals.get("priority") == "2":
+                vals["request_late"] = fields.Datetime.from_string(
+                    vals.get("request_early")
+                ) + timedelta(days=1)
+            elif vals.get("request_early") and vals.get("priority") == "3":
+                vals["request_late"] = fields.Datetime.from_string(
+                    vals.get("request_early")
+                ) + timedelta(hours=8)
```

Comment:
The `_compute_request_late()` function is only used for tests, but actually the value of the `request_late` field is defined on creation. I think it would be best to remove it completely, what do you think?